ecmascript/transforms/react/src/jsx/mod.rs

```diff
@@ -1130,42 +1130,28 @@ fn to_prop_name(n: JSXAttrName) -> PropName {
 
 #[inline]
 fn jsx_text_to_str(t: JsWord) -> JsWord {
-    static SPACE_NL_START: Lazy<Regex> =
-        Lazy::new(|| Regex::new("^[\t'\n\x0C\r ]*\n[\t'\n\x0C\r ]*").unwrap());
-    static SPACE_NL_END: Lazy<Regex> =
-        Lazy::new(|| Regex::new("[\t'\n\x0C\r ]*\n[\t'\n\x0C\r ]*$").unwrap());
-
-    if t == *" " {
-        return t;
-    }
-
-    if !t.contains(' ') && !t.contains('\n') {
-        return t;
-    }
-
-    let s = SPACE_NL_START.replace_all(&t, "");
-    let s = SPACE_NL_END.replace_all(&s, "");
-    let need_leading_space = s.starts_with(' ');
-    let need_trailing_space = s.ends_with(' ');
-
-    let mut buf = String::from(if need_leading_space { " " } else { "" });
-
-    for (last, s) in s
-        .split(|c: char| c != '\u{a0}' && c.is_ascii_whitespace())
-        .filter(|s| !s.is_empty())
+    static SPACE_START: Lazy<Regex> = Lazy::new(|| Regex::new("^[ ]+").unwrap());
+    static SPACE_END: Lazy<Regex> = Lazy::new(|| Regex::new("[ ]+$").unwrap());
+    t.lines()
+        .map(|s| s.replace('\t', " "))
+        .enumerate()
+        .filter(|(_, v)| v.len() != 0)
         .identify_last()
```

Comment:
This seems like an inefficient code, as it allocates lots of `String`s.
I think using a buffer is preferrable.